interpreter/cling/lib/Interpreter/LookupHelper.cpp

```diff
@@ -722,7 +722,10 @@ namespace cling {
                         // in invalid state. We should be unloading all of them, i.e. inload the
                         // current (possibly nested) transaction.
                         auto *T = const_cast<Transaction*>(m_Interpreter->getCurrentTransaction());
-                        m_Interpreter->unload(*T);
+                        // Must not unload the Transaction, which might delete it:
+                        // an RAII in an upper call frame might point to it! Instead, just
+                        // mark it as "erroneous" which will unload it in due time.
+                        T->setIssuedDiags(Transaction::kErrors);
```

Comment:
Can a Transaction be used with an RAII?  If it can, how it is delete/unloaded in that case (no RAII)?

Reply:
Line 541 has an RAII, this line *will* have one by definition. (And again, the subsequent PR will ensure that we will not break this fragile context. But I will likely not backport it to 6.24.)